client/cmd.go

```diff
@@ -99,6 +99,11 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 		clientCtx = clientCtx.WithHomeDir(homeDir)
 	}
 
+	if !clientCtx.Simulate || flagSet.Changed(flags.FlagDryRun) {
+		dryRun, _ := flagSet.GetBool(flags.FlagDryRun)
+		clientCtx = clientCtx.WithSimulation(dryRun)
+	}
```

Comment:
1. in the if, we we use `flagSet.Changed(flags.FlagDryRun)` rather than `flagSet.GetBool(flags.FlagDryRun)`?
2. if we are in simulate, should we assume that `dryRun == true`?

Maybe we should write it as (note removal of negation before clientCtx.Simulate):
```
if clientCtx.Simulate || flagSet.GetBool(flags.FlagDryRun) {
	clientCtx = clientCtx.WithSimulation(true)
}
```

Reply:
ping

Reply:
> 1. in the if, we we use flagSet.Changed(flags.FlagDryRun) rather than flagSet.GetBool(flags.FlagDryRun)

I believe that Gianguido has just replicated a pattern that he had found across the `client` package. I honestly neither know nor understand why the `Changed()` method is used. It makes very little sense to me.

Reply:
I think the function godoc string explains it though:

```

// ReadPersistentCommandFlags returns a Context with fields set for "persistent"
// or common flags that do not necessarily change with context.
//
// Note, the provided clientCtx may have field pre-populated. The following order
// of precedence occurs:
//
// - client.Context field not pre-populated & flag not set: uses default flag value
// - client.Context field not pre-populated & flag set: uses set flag value
// - client.Context field pre-populated & flag not set: uses pre-populated value
// - client.Context field pre-populated & flag set: uses set flag value
```

Reply:
yeah, there is so much code copy paste - would be good in a future to put that patterns in a function (no need to copy paste same pattern)